pkg/channel/consolidated/reconciler/dispatcher/kafkachannel.go

```diff
@@ -157,6 +157,16 @@ func (r *Reconciler) ObserveKind(ctx context.Context, kc *v1beta1.KafkaChannel)
 	return r.syncDispatcher(ctx)
 }
 
+func (r *Reconciler) FinalizeKind(ctx context.Context, kc *v1beta1.KafkaChannel) pkgreconciler.Event {
+	logging.FromContext(ctx).Debugw("FinalizeKind for channel", zap.String("channel", kc.Name))
+	return r.finalizeChannel(ctx, kc)
+}
+
+func (r *Reconciler) ObserveFinalizeKind(ctx context.Context, kc *v1beta1.KafkaChannel) pkgreconciler.Event {
```

Comment:
FWIW, this is "wrong", see https://github.com/knative/pkg/issues/2021

Reply:
Isn't adding both fine? The `CleanupChannel` method is idempotent. If it's going to cause issues, I can remove it

Reply:
I've removed both FinalizeKind and ObserveFinalizeKind and replaced with a specific handler on the informer